ui-admin/src/components/forms/TextInput.tsx

```diff
@@ -26,13 +26,15 @@ export const TextInput = (props: TextInputProps) => {
         {label}
       </label>
       <input
+        type="text"
         {...inputProps}
         aria-describedby={description ? descriptionId : undefined}
         aria-disabled={disabled}
         className={classNames('form-control', { disabled }, className)}
         disabled={undefined}
         id={inputId}
-        type="text"
+        // Allow value to be undefined without triggering a React warning about uncontrolled input.
+        value={value || ''}
```

Comment:
generally recommend using `??` instead of `||` since ?? doesn't have edge cases around handling of 0 and empty strings (not that it matters for this specific case)
```suggestion
        value={value ?? ''}
```

Reply:
Done in https://github.com/broadinstitute/juniper/pull/448/commits/88cfdef6a0b2ae7f13631cdaa1b6b55e268397b9. That allows simplifying NumberInput too.